addons/sale_timesheet/tests/test_upsell_warning.py

```diff
@@ -86,3 +86,70 @@ def test_display_upsell_warning(self):
 
         # 5) Check if the SO has an 'sale.mail_act_sale_upsell' activity.
         self.assertEqual(len(so.activity_search(['sale.mail_act_sale_upsell'])), 1, 'A upsell warning should appear in the SO.')
+
+    def test_display_upsell_warning_2(self):
```

Comment:
Your test does not fail without your fix so it means the test does not really reproduce the issue. 🤔

Reply:
@xavierbol ,
Please check the changes made in the test whenever you have time.
Thanks!!